### ISSUE TITLE
Extracting only ground truth masks from training dataset (to build a confusion matrix)?

### ISSUE THREAD

**Issue by eliasm56** [qubvel/segmentation_models.pytorch#382]
Hi everyone,

I have successfully trained a U-Net using this library with the CamVid example provided for this repo. Now I am trying to put together some evaluation assessments (plots, graphs, etc.). I am trying to build a confusion matrix to compare the ground truth masks and predicted masks from the training set. Does anyone know how to extract only the masks from the training dataset? Here is how the dataset object was created (taken from the CamVid example and modified):

`# Create helper classes for data preprocessing and augmentation.
class Dataset(BaseDataset):
    
    def __init__(
            self, 
            images_dir, 
            masks_dir, 
            augmentation=None, 
            preprocessing=None,
    ):
        self.ids = os.listdir(images_dir)
        self.images_fps = [os.path.join(images_dir, image_id) for image_id in self.ids]
        self.masks_fps = [os.path.join(masks_dir, image_id) for image_id in self.ids]
        
        self.augmentation = augmentation
        self.preprocessing = preprocessing
    
    def __getitem__(self, i):
        
        # Read and resize images and masks to desired resolution. *Interpolation argument must be set to nearest-neighbor
        # to preserve ground truth.
        image = cv2.imread(self.images_fps[i])
        image = cv2.resize(image, (256, 256))
        mask = cv2.imread(self.masks_fps[i], 0)
        mask = cv2.resize(mask, (256, 256), interpolation = cv2.INTER_NEAREST)
        
        # One-hot encode masks. In this case, low-center polygons have been assigned a pixel value of 119
        # and high-center polygons have been assigned a pixel value of 238 on grayscale. To prepare the masks
        # for one-hot encoding, these pixel values must be re-assigned to 1 and 2, background will remain 0.
        mask[mask == 119] = 1
        mask[mask == 238] = 2
        masks = tf.one_hot(mask, 3, axis = 0)
        mask = np.stack(masks, axis=-1).astype('float')
      
        
        # Apply augmentations
        if self.augmentation:
            sample = self.augmentation(image=image, mask=mask)
            image, mask = sample['image'], sample['mask']
        
        # Apply preprocessing
        if self.preprocessing:
            sample = self.preprocessing(image=image, mask=mask)
            image, mask = sample['image'], sample['mask']
            
        return image, mask
        
    def __len__(self):
        return len(self.ids)`

Furthermore, here is how I created a tensor holding all the predicted masks from training:

`@torch.no_grad()
def get_all_preds(model, loader):
    all_preds = torch.tensor([])
    for batch in loader:
        images, labels = batch

        preds = model(images)
        all_preds = torch.cat(
            (all_preds, preds)
            ,dim=0
        )
    return all_preds`

**Comment by eliasm56** [comment]
For some reason the code I attached got formatted in a weird manner. First code block starts at the "#" and second block starts at the "@"
